src/main/java/duke/ui/BotUI.java

```diff
@@ -0,0 +1,162 @@
+
+import duke.storage.TaskRecords;
+import duke.task.Task;
+
+/**
+ * Represents the user interface of duke chatBot.
+ * Contains necessary messages to respond to the user.
+ * Contains method that format the responses to more interactive.
+ */
+public class BotUI {
+
+    private String botSpeak(String phrase) {
+        return MESSAGE_BOT_RESPONSE + phrase;
+    }
+
+    public static String userSpeak() {
+        return MESSAGE_USER_SAY;
+    }
+    /**
+     * Reads the user input.
+     * @return the String of user input.
+     */
+    public String readCommand() {
+        Scanner scn = new Scanner(System.in);
+        return scn.nextLine();
+    }
+    /**
+     * Returns hello message.
+     * @return hello message from Duke.
+     */
+    public String sayHello() {
+        return botSpeak(MESSAGE_HI);
+    }
+    /**
+     * Returns bye message.
+     * @return hello message from Duke.
+     */
+    public String sayBye() {
+        return botSpeak(MESSAGE_BYE);
+    }
+    /**
+     * Returns the tasks contain in the TaskRecords.
+     * @param taskList TaskRecords that store the tasks registered by user.
+     * @return Numbering-formatted String of the tasks in the task list.
+     */
+    public String showList(TaskRecords taskList) {
+        ArrayList<Task> lst = taskList.getList();
+        if (lst.size() == 0) {
+            return botSpeak(MESSAGE_NOTHING_IN_LIST);
+        } else {
+            StringBuilder lstFormat = new StringBuilder();
+            for (int i = 1; i <= lst.size(); i++) {
+                lstFormat.append(i).append(". ").append(lst.get(i - 1).toString());
+                lstFormat.append((i == lst.size()) ? "" : "\n");
+            }
+            return botSpeak(lstFormat.toString());
+        }
+    }
+    /**
+     * Returns complete status of the task.
+     * @param task Task that have been mark after the user command.
+     * @return The String of completed task.
+     */
+    public String informMarkStatus(Task task) {
+        String response = botSpeak((task.isDone()) ? String.format(MESSAGE_MARK_SUCCESS, task)
+               : String.format(MESSAGE_UNMARK_SUCCESS, task));
+        return response;
+    }
+    /**
+     * Returns task added into the TaskRecords.
+     * @param task Task that have been added by user.
+     * @param taskList the updated TaskRecords to show user the updated number of tasks in it.
+     * @return The String of task added.
+     */
+    public String addStatus(TaskRecords taskList, Task task) {
+        ArrayList<Task> lst = taskList.getList();
+        return botSpeak(String.format(MESSAGE_ADD_SUCCESS, task, lst.size()));
+    }
+    /**
+     * Returns task removed from the TaskRecords.
+     * @param task Task that have been added by user.
+     * @param taskList the updated TaskRecords to show user the updated number of tasks in it.
+     * @return The String of task added.
+     */
+    public String successRemoved(TaskRecords taskList, Task task) {

```

Comment:
Method name can be confusing to readers unfamiliar with the bot. Perhaps a more intuitive name could be used here.

Reply:
Okay, will try to think more about other code reader perspectives